src/main/scala/com/udemy/statistics/distribution/package.scala

```diff
@@ -0,0 +1,36 @@
+/*
+Copyright 2018 Udemy, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+ */
+
+package com.udemy.statistics
+
+package object distribution {
+  case class Interval(lower: Double, upper: Double)
+
+  case class Percentile(value: Double) {
+    require(value >= 0 && value <= 1, "percentile must be a value between 0 and 1")
+
+    def bounds: Interval = {
+      val lower = (1 - value) / 2
+      val upper = value + lower
+      Interval(lower, upper)
+    }
+  }
+
+  object CredibleInterval extends Enumeration {
```

Comment:
ugh, i've always thought scala's enum implementation is confusing. 
curious why you're implementing this now with only one element to your enum

Reply:
Good question. Because there are multiple types of credible intervals (CIs) I want to be sure the consumer of the CI functions knows which CI we're returning. And we hope to support other CIs in the future. If we don't include this now we'd have to either work on the assumption that this is the type of CI the consumer wants or encode the type of CI in the function name. But I prefer to have fewer functions and encode types elsewhere in the signature, in this case in the arguments.